tracing/span.go

```diff
@@ -39,153 +40,89 @@ func (st SpanType) String() string {
 	}
 }
 
-type contextKey int
-
-const (
-	serverSpanKey contextKey = iota
-	childSpanKey
-)
-
-// FlagMask values.
-//
-// Reference: https://github.com/reddit/baseplate.py/blob/1ca8488bcd42c8786e6a3db35b2a99517fd07a99/baseplate/observers/tracing.py#L60-L64
-const (
-	// When set, this trace passes all samplers.
-	FlagMaskDebug int64 = 1
-)
-
-const (
-	counterKeyPrefix   = "counter."
-	baseplateComponent = "baseplate"
-)
-
-// Span defines a tracing span.
-type Span struct {
-	Name string
-
+// Trace holds the trace data for a Span and publishes ZipkinSpans.
+type Trace struct {
```

Comment:
Do we use `Trace` as the top level `Span`, or do we only use it to hold internal state? If it's only for internal state then I don't think we want to export it.

Reply:
Lemme fiddle around with this more, maybe we can just not export it.